### DISCUSSION TITLE
Is those functions possible to do in Saleor?

### DISCUSSION THREAD

**Discussion by HarunKilic** [saleor/saleor#9292] [Q&A] [answered]
Hi guys,

I am currently developing ecommerce for B2B, and Saleor seems pretty nice and promising. 
But i have some questions about extendability without customising core. 
I want to build an app or plugin for ERP synchronisation so that Data comes from ERP instead of adding from web dashboard. 

The functionality i am looking for: 

### Companies - A company is the key for this shop
- Users can join multiple companies (The login should be by using Users own credentials and not companies) 
- Users will only see prices on the products if they're in a company.
- Users will only be able to order if they're in a company
- Company will have some custom metadatas 

I was thinking about mapping Companies as Channels, but not sure if its a good idea. As there will be 3000+ companies and linking all items per company will be too much.

### Orders
- Orders will be created by tree required keys: User + Company + Store (Warehouse or Channels)

### Pricing
- There will be different pricing per company. So product can have different price for Company A and Company B. 
- There will also be price tiering, meaning that buying 5 pieces will have different price. 

### Products
- Same product can be available in different stores, and some of them only in some stores. 
- Products have unit of measures. A product has eg. 1piece, 4pieces (box)
   - An unit of measure will have unique ean/barcode. 
   - An unit of measure might not be sellable. Eg. 1 Piece will not be buyable but only the box will. 
 
Those are some of the functionalities i want from an ecommerce. 

So the question is: 
Is it possible to make those configurations in Saleor trough app or plugin?

**Comment by karolkielecki** [comment, marked as answer]
Hi @HarunKilic. It's not possible to do that through an app or plugin. Saleor doesn't support different pricing based on quantity. IMO, your use case would require heavy core modification, adding new models, modifying the logic of orders, pricing, permissions.

**Reply by HarunKilic** [reply]
Hi @karolkielecki 

Thank you for your response. I hope Saleor will fit the case some time in the future as i see great opportunity in the project :)
